woocommerce/woocommerce
# Integrate Search component into in-app marketplace header

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Integrating search component into in-app marketplace header.

- Changed classNames in `Search` component.
- Moved styles from `header.scss` to `search.scss`.
- Deleted `HeaderSearch` and `HeaderSearchButton` components, no longer needed.
- Removed wrapper element around search form.
- Corrected marketplace API address in `Search` component.
- Using state to remember search term, instead of getting value of search input.
- Added condition to clear search input if Escape is pressed while it's focused.
- Not running search if search term is falsy.
- Changed label of "Extensions" tab to "Browse" – we'll do this for the moment, as we have repetition with the page title `Extensions`.

Closes [17693-gh-Automattic/woocommerce.com](17693-gh-Automattic/woocommerce.com).



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out this branch and build WooCommerce Admin on your local environment: `cd plugins/woocommerce-admin` then `nvm use && pnpm start`. (If you get a fatal error, refresh the Composer autoload: `cd plugins/woocommerce` and `composer dump-autoload`.)
2. Go to the marketplace (if you're using `wp-env`, it's `http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions`).
3. Open the network tab of dev tools, and filter by `wccom-extensions`.
4. Check that the search input looks OK on different viewports.
5. Enter a search term and press your Enter key. Notice that the component makes a request to the extensions API and gets a response.
6. Click the search icon in the input. The component should make another request.
7. Click your Escape key. The term in the input should clear.
8. Click the search icon in the input. The component should not make a request.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
